Project/DAQ_System/Program/Src/Sensor/ECU/PE3/pe3.hpp

```diff
@@ -14,12 +14,186 @@
 #ifndef ECU_PE3_PE3_H
 #define ECU_PE3_PE3_H
 
+// Standard Libraries
+#include <inttypes.h>
+
+#include <vector>
+#include <array>
+#include <memory>
+
+// DFR Custom Dependencies
+#include "iecu.hpp"
+#include "../../../Platform/CAN/Interfaces/ican.hpp"
+
+// Internal ECU Messages
+#include "Frames/frame_pe1.hpp"
+#include "Frames/frame_pe2.hpp"
+#include "Frames/frame_pe3.hpp"
+#include "Frames/frame_pe4.hpp"
+#include "Frames/frame_pe5.hpp"
+#include "Frames/frame_pe6.hpp"
+#include "Frames/frame_pe7.hpp"
+#include "Frames/frame_pe8.hpp"
+#include "Frames/frame_pe9.hpp"
+#include "Frames/frame_pe10.hpp"
+#include "Frames/frame_pe11.hpp"
+#include "Frames/frame_pe12.hpp"
+#include "Frames/frame_pe13.hpp"
+#include "Frames/frame_pe14.hpp"
+#include "Frames/frame_pe15.hpp"
+#include "Frames/frame_pe16.hpp"
+
 namespace sensor {
 
-class Pe3 {
+#define FramePe1Id		0x0CFFF048
+#define FramePe2Id		0x0CFFF148
+#define FramePe3Id		0x0CFFF248
+#define FramePe4Id		0x0CFFF348
+#define FramePe5Id		0x0CFFF448
+#define FramePe6Id		0x0CFFF548
+#define FramePe7Id		0x0CFFF648
+#define FramePe8Id		0x0CFFF748
+#define FramePe9Id		0x0CFFF848
+#define FramePe10Id		0x0CFFF948
+#define FramePe11Id		0x0CFFFA48
+#define FramePe12Id		0x0CFFFB48
+#define FramePe13Id		0x0CFFFC48
+#define FramePe14Id		0x0CFFFD48
+#define FramePe15Id		0x0CFFFE48
+#define FramePe16Id		0x0CFFD048
+
+class Pe3 : IEcu {
```

Comment:
Caught a bug. I left out the `public` keyword when inheriting the `IEcu` interface. This was giving me issue with pointer arithmetic at initialization. Here is an [example](https://www.reddit.com/r/cpp_questions/comments/18un06p/error_error_conversion_from_shared_ptr_to/).